app/modules/intelligence/provider/provider_service.py

```
@@ -1,39 +1,55 @@
     CREWAI = "CREWAI"
     LANGCHAIN = "LANGCHAIN"
 
-
 class ProviderService:
     def __init__(self, db, user_id: str):
+        """Initialize ProviderService with database session and user ID."""
+        logger.info(f"Initializing ProviderService for user_id: {user_id}")
         self.db = db
         self.llm = None
         self.user_id = user_id
         if os.getenv("isDevelopmentMode") != "enabled":
```

Ruff (0.8.2)

Use capitalized environment variable `ISDEVELOPMENTMODE` instead of `isDevelopmentMode`

(SIM112)